argopy/fetchers.py

```diff
@@ -205,3 +208,124 @@ def to_xarray(self, **kwargs):
+
+        # Load backend access points:
+        if self._backend not in AVAILABLE_BACKENDS:
+            raise ValueError("Data fetcher '%s' not available" % self._backend)
+        else:
+            Fetchers = AVAILABLE_BACKENDS[self._backend]
+
+        # Auto-discovery of access points for this fetcher:
+        # rq: Access point names for the facade are not the same as the access point of fetchers
+        self.valid_access_points = ['float', 'region']
+        self.Fetchers = {}
+        for p in Fetchers.access_points:
+            if p == 'wmo':  # Required for 'profile' and 'float'                
+                self.Fetchers['float'] = Fetchers.IndexFetcher_wmo
+            if p == 'box':  # Required for 'region'
+                self.Fetchers['region'] = Fetchers.IndexFetcher_box
+
+        # Init sub-methods:
+        self.fetcher = None
+        self.fetcher_options = {**fetcher_kwargs}
+        self.postproccessor = self.__empty_processor                
+
+    def __repr__(self):
+        if self.fetcher:
+            summary = [self.fetcher.__repr__()]
+            summary.append("User mode: %s" % self._mode)
+        else:
+            summary = ["<datafetcher 'Not initialised'>"]
+            summary.append("Fetchers: 'float' or 'region'")
+            summary.append("User mode: %s" % self._mode)
+        return "\n".join(summary)
+
+    def __empty_processor(self, xds):
+        """ Do nothing to a dataset """
+        return xds
+
+    def float(self, wmo):
+        """ Load index for one or more WMOs """
+        if 'float' in self.Fetchers:
+            self.fetcher = self.Fetchers['float'](WMO=wmo, **self.fetcher_options)
+            print("Float index initialised")
+        else:
+            raise InvalidFetcherAccessPoint("'float' not available with '%s' backend" % self._backend)        
+        return self    
+
+    def region(self, box):
+        """ Load index for a rectangular region, given latitude, longitude, and possibly time bounds """
+        if 'region' in self.Fetchers:
+            self.fetcher = self.Fetchers['region'](box=box, **self.fetcher_options)
+            print("Box index initialised")
+        else:
+            raise InvalidFetcherAccessPoint("'region' not available with '%s' backend" % self._backend)                     
+        return self        
+
+    def to_dataframe(self, **kwargs):
+        """ Fetch index and return pandas.Dataframe """
+        pddf = self.fetcher.to_dataframe(**kwargs)        
+        return pddf        
+
+    def to_xarray(self, **kwargs):
+        """ Fetch index and return xr.dataset """
+        pddf = self.fetcher.to_dataframe(**kwargs) 
+        ds = pddf.to_xarray()       
+        return ds            
+
+    def to_csv(self,file='output_file.csv'):
+        """ Fetch index and return csv """
+        idx=self.to_dataframe()
+        idx.to_csv(file)   
+        print('file saved to : '+file)             
```

Comment:
Return nothing or the filename, let pandas decides
```suggestion
        return idx.to_csv(file)
```